### PR TITLE
add padding to `flip` and `shift` middleware

### PR DESCRIPTION
**Pull request by mayank99**
## Changes

This is a small change that stems from #2187. I've specified a small [`padding`](https://floating-ui.com/docs/detectOverflow#padding) value in the edge detection middlewares so that the tooltips/popovers are not touching the edge of the window and have a little more breathing room.

## Testing

Verified that the padding value is working.

| Before | After |
| --- | --- |
|  |  |

Updated failing visual test images.

## Docs

Added `patch` changeset.

### CODE REVIEW

File: packages/itwinui-react/src/core/Tooltip/Tooltip.tsx [hunk 1]
```diff
@@ -177,8 +177,8 @@ const useTooltip = (options: TooltipOptions = {}) => {
           middleware.offset !== undefined
             ? offset(middleware.offset)
             : offset(4),
-          middleware.flip && flip(),
-          middleware.shift && shift(),
+          middleware.flip && flip({ padding: 4 }),
+          middleware.shift && shift({ padding: 4 }),
           middleware.size && size(),
```
[the remark this answers is not in the document]
Comment (mayank99) [reply packages/itwinui-react/src/core/Tooltip/Tooltip.tsx:182]: These are the only two middlewares that are enabled by default. I don't think `size` even does anything in `Tooltip`, but I have another PR #2190 up where I'm using `size` in `Popover`.

**Review by smmr-dn — APPROVED** [review]
I had the same question as Rohan about the padding for other middleware option. Other than that, this looks good to me!

File: packages/itwinui-react/src/core/Tooltip/Tooltip.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (mayank99) [reply packages/itwinui-react/src/core/Tooltip/Tooltip.tsx]: sure, done [`7298540`](https://github.com/iTwin/iTwinUI/pull/2189/commits/72985400f09647250eca0c5847b23d1b7f405345)

File: packages/itwinui-react/src/core/Tooltip/Tooltip.tsx [hunk 2]
```diff
@@ -177,12 +177,12 @@ const useTooltip = (options: TooltipOptions = {}) => {
           middleware.offset !== undefined
             ? offset(middleware.offset)
             : offset(4),
-          middleware.flip && flip(),
-          middleware.shift && shift(),
-          middleware.size && size(),
-          middleware.autoPlacement && autoPlacement(),
+          middleware.flip && flip({ padding: 4 }),
```
[the remark this answers is not in the document]
Comment (mayank99) [reply packages/itwinui-react/src/core/Tooltip/Tooltip.tsx:180]: I think it's fine as-is. `4` is a simple value and these middlewares are close enough to each other that duplication is not really a problem.

**Pull request merged**